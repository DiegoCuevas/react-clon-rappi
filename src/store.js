import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import thunk from "redux-thunk";
import logger from "redux-logger";
import offlineSync from "./midlewares/offline-sync";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

if (localStorage.getItem("version") !== process.env.REACT_APP_VERSION) {
  localStorage.removeItem("state");
}
const preloadedState = JSON.parse(localStorage.getItem("state") || "{}");

const store = createStore(
  reducer,
  // preloadedState,
  // composer(applyMiddleware(thunk, logger, offlineSync))
  composer(applyMiddleware(thunk, logger))
);

export default store;
