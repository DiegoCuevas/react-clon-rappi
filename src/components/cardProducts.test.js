import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../store";
import CardProduct from "./cardProduct";

test("cardProduct component", () => {
  const restaurant = {
    id: 1,
    name: "Restaurant-1"
  };

  const product = {
    name: "Product-1"
  };

  const { asFragment } = render(
    <Provider store={store}>
      <CardProduct idRestaurant={restaurant.id} product={product} />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
