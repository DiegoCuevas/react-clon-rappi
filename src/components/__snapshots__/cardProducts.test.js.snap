// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardProduct component 1`] = `
<DocumentFragment>
  .emotion-7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
  margin: 10px 5px;
  min-width: 30%;
}

@media (max-width:768px) {
  .emotion-7 {
    min-width: 100%;
  }
}

.emotion-6 {
  border-radius: .5em;
  box-sizing: border-box;
  border: 1px solid #fff;
  padding: 15px;
  width: 100%;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.25s;
  transition-duration: 0.25s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.emotion-6:hover {
  cursor: pointer;
  border: 1px solid #e5edef;
  box-shadow: -1px 2px 5px 0px rgba(0,0,0,.12);
}

@media (max-width:768px) {
  .emotion-6 {
    border-radius: 0px;
    border-top: 1px solid #e5edef;
    border-buttom: 1px solid #e5edef;
  }
}

.emotion-0 {
  position: relative;
  width: 124px;
}

.emotion-5 {
  padding: 10px;
}

.emotion-1 {
  font-size: 16px;
  color: #333;
  font-weight: 400;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: block;
}

.emotion-2 {
  font-size: 12px;
  color: #9faab7;
  margin: 0;
  min-height: 12px;
  text-overflow: ellipsis;
  display: block;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-4 {
  font-size: 12px;
  line-height: 20px;
  color: #333;
  text-align: right;
}

.emotion-3 {
  color: #9faab7;
  display: inline-block;
  vertical-align: top;
  font-size: 5px;
  padding: 0 5px;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-0"
      >
        <img
          alt="restaurant"
          src="https://images.rappi.pe/restaurants_background/1839-1556640864.png?d=124x124&e=webp"
        />
      </div>
      <div
        class="emotion-5"
        title="Add to cart"
      >
        <span
          class="emotion-1"
        >
          Product-1
        </span>
        <span
          class="emotion-2"
        />
        <span
          class="emotion-4"
        >
          Envio S/.  
          <span
            class="emotion-3"
          >
            ‚óè
          </span>
           min
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
